src/subscriptions/subscriptionBackfill.ts

```diff
@@ -129,15 +135,29 @@ export function makeBackfiller(jsonRpcSenders: JsonRpcSenders) {
     if (fromBlockInclusive >= toBlockExclusive) {
       return [];
     }
-    const batchParts: BatchPart[] = [];
+    let batchParts: BatchPart[] = [];
+    const headEventBatches: Array<Promise<BlockHead[]>> = [];
     for (let i = fromBlockInclusive; i < toBlockExclusive; i++) {
       batchParts.push({
         method: "eth_getBlockByNumber",
         params: [toHex(i), false],
       });
+      if (batchParts.length % MAX_BATCH_SIZE === 0) {
+        headEventBatches.push(jsonRpcSenders.sendBatch(batchParts));
+        batchParts = [];
+      }
     }
-    const heads = await jsonRpcSenders.sendBatch(batchParts);
-    return heads.map(toNewHeadsEvent);
+
+    if (batchParts.length > 0) {
+      headEventBatches.push(jsonRpcSenders.sendBatch(batchParts));
+    }
+
+    const batchedBlockHeads = await Promise.all(headEventBatches);
```

Comment:
do we need to do any error handling here?

Reply:
The jsonRpcSenders already perform retries with backoff. If all the retries fail, the backfill fails silently. the backfill itself is also a silent background process, so the end user shouldn't see any errors.